### PR TITLE
[362] [scaffold] Pending Transaction Details

### LINKED ISSUE #362: [scaffold] Pending Transaction Details
As a User I would like to see the details available for a pending transaction:

pending confirmation and estimated time
transaction id
value amount
Fee
memo (if available)
reply-to compliance

total amount

Items that can be copied to clipboard:

memo
transaction ID

Reference: (look and feel not needed on this story)

### LINKED ISSUE #363: [scaffold] Failed (pending) transaction details
As a User I would like to see the details available for a failed pending transaction:

transaction creation estimated time
transaction id (if available)
value amount
Fee
memo (if available)
reply-to compliance
total amount
Items that can be copied to clipboard:
failure reason
memo
transaction ID

Reference: (look and feel not needed on this story)

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #362 
Closes #363 

- transaction detail has been cut into pieces that can be composed to various versions of the detail
- handling all 4 cases (sent, received, pending, failed)
- confirmations counting
- pending transaction status fixed
- tests fixed
- failed transactions

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift [hunk 1]
```diff
@@ -2,72 +2,160 @@ import SwiftUI
...
-                            } label: {
-                                Text("reply now")
-                                    .padding(5)
-                                    .overlay(
-                                        RoundedRectangle(cornerRadius: 6)
-                                            .stroke(Asset.Colors.Text.transactionDetailText.color, lineWidth: 1)
-                                    )
-                            }
-                        }
+    func plainTwoColumnText(left: String, right: String, mark: RowMark = .neutral) -> some View {
```
Comment (pacu) [inline secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift:85]: Shouldn't these be `@ViewBuilder` functions?

File: secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift [hunk 2]
```diff
@@ -112,33 +194,44 @@ extension TransactionDetailView {
...
 struct TransactionDetailRow: ViewModifier {
-    let tint: Color
+    let mark: TransactionDetailView.RowMark
     let textColor: Color
     let backgroundColor: Color
 
     func body(content: Content) -> some View {
-        content
+        let markColor: Color
+        switch mark {
```
Comment (pacu) [inline secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift:206]: I think this is oddly indented, isn't it?

**Review by pacu — CHANGES_REQUESTED** [review]
question on `@ViewBuilder` functions and some formatting remarks

File: secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift (hunk 1 above)
Comment (LukasKorba) [reply secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift:85]: No need to wrap it with @ViewBuilder. All functions are returning exact View type so SwiftUI doesn't need any helper (ViewBuilder, wrap to Group, wrap to AnyView or new struct View). We essentially have other views full or composable comonents 
```
// either
var header: some View

// or
func header() -> some View
```

As long as var or func is returning 1 specific type, SwiftUI is ok. When if/switch statements, one of the 4 mentioned solutions would be needed.

File: secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift (hunk 2 above)
Comment (LukasKorba) [reply secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift:206]: I pulled out the color switch to a private function which simplified the body of the modifier

**Review by pacu — APPROVED** [review]

**Pull request merged**